modules/core/src/main/java/com/google/refine/model/changes/ColumnRenameChange.java

```diff
@@ -89,18 +89,176 @@ static public Change load(LineNumberReader reader, Pool pool) throws Exception {
         String line;
         while ((line = reader.readLine()) != null && !"/ec/".equals(line)) {
             int equal = line.indexOf('=');
-            CharSequence field = line.subSequence(0, equal);
+            if (equal == -1) {
+                // This might be a continuation line from corrupted legacy data
+                continue;
+            }
+
+            String field = line.substring(0, equal);
             String value = line.substring(equal + 1);
 
             if ("oldColumnName".equals(field)) {
-                oldColumnName = value;
+                oldColumnName = processFieldValue(value, reader);
             } else if ("newColumnName".equals(field)) {
-                newColumnName = value;
+                newColumnName = processFieldValue(value, reader);
             }
         }
 
-        ColumnRenameChange change = new ColumnRenameChange(oldColumnName, newColumnName);
+        return new ColumnRenameChange(oldColumnName, newColumnName);
+    }
+
+    /**
+     * Process a field value, handling both new escaped format and legacy format. For legacy format, handle potential
+     * corruption from newline characters.
+     */
+    private static String processFieldValue(String value, LineNumberReader reader) throws IOException {
+        if (containsEscapeSequences(value)) {
+            return unescape(value);
+        } else {
+            // Legacy format - might be corrupted if original value contained newlines
+            return handleLegacyValue(value, reader);
+        }
```

Comment:
I need to think about this more. I think it would be better to use the version information written to the change file, but I'm not sure how reliable it is and how difficult it would be to make use of.